lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelHistoryTest.java

```diff
@@ -1355,4 +1355,108 @@ public void run() {
 				rxAbly.close();
 		}
 	}
+
+	/**
+	 * Connect twice to the service, each using the default (binary) protocol.
+	 * Publish messages on one connection to a given channel; while in progress,
+	 * attach the second connection to the same channel and verify a message
+	 * history up to the point of attachment can be obtained.
+	 */
+	@Test
+	public void channelhistory_until_attach() {
+		AblyRealtime txAbly = null, rxAbly = null;
+		try {
+			io.ably.lib.test.common.Setup.TestVars testVars = Setup.getTestVars();
+			ClientOptions txOpts = testVars.createOptions(testVars.keys[0].keyStr);
+			txAbly = new AblyRealtime(txOpts);
+			ClientOptions rxOpts = testVars.createOptions(testVars.keys[0].keyStr);
+			rxAbly = new AblyRealtime(rxOpts);
+			String channelName = "persisted:channelhistory_until_attach";
+
+			/* create a channel */
+			final Channel txChannel = txAbly.channels.get(channelName);
+			final Channel rxChannel = rxAbly.channels.get(channelName);
+
+			/* attach sender */
+			txChannel.attach();
+			(new ChannelWaiter(txChannel)).waitFor(ChannelState.attached);
+			assertEquals("Verify attached state reached", txChannel.state, ChannelState.attached);
+
+			/* publish messages to the channel */
```

Comment:
Might be worth modifying the comment to state that messages are published over a period of 5 seconds, and the attach request on the receiving channel is made at 2s.  However, this does seem like we're relying too much on everything to be synchronised.  Why not simply publish 25 messages before rxChannel is attached and wait for the 25 publishes to be completed, then publish 25 messages after rxChannel is attached, in this way there is NO chance that an attach delay for example could cause this test to fail.  Then once the 25 messages are received on rxChannel, do a history request with untilAttach and expect exactly 25 messages with IDs 0 -> 25.  This seems simpler to me and certainly a lot less brittle.